src/components/PlaybackControls/index.tsx

```diff
@@ -69,33 +77,14 @@ const PlayBackControls = ({
         }
     };
 
-    const units = ["s", "ms", "\u03BCs", "ns"];
     const roundNumber = (num: number) => parseFloat(Number(num).toPrecision(3));
-    const roundedTime = time ? roundNumber(time * 1000 ** unitIndex) : 0;
-    const roundedLastFrameTime = roundNumber(lastFrameTime * 1000 ** unitIndex);
-
-    // Calculates display unit when lastFrameTime is updated, i.e., when a new trajectory is loaded
-    useEffect(() => {
-        if (!lastFrameTime) return;
-        /*
-        All incoming times are in seconds, but we want to determine the best unit for displaying.
-        
-        Here we determine the most appropriate unit by calculating how many times (rounded up) the inverse of
-        lastFrameTime can divide by 1000. Math.log(x) / Math.log(1000) is the same as log base 1000 of x:
-        https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log/#Examples
-        */
-        let index = Math.ceil(Math.log(1 / lastFrameTime) / Math.log(1000));
-
-        // Handle very small values (use ns if lastFrameTime is less than 1 ns)
-        if (index >= units.length) {
-            index = units.length - 1;
-            // Handle very large values (use s if lastFrameTime is greater than 1000 s)
-        } else if (index < 0) {
-            index = 0;
-        }
+    let roundedTime = 0;
+    let roundedLastFrameTime = 0;
 
-        setUnitIndex(index);
-    }, [lastFrameTime]);
+    if (timeUnits) {
+        roundedTime = time ? roundNumber(time * timeUnits.magnitude) : 0;
+        roundedLastFrameTime = roundNumber(lastFrameTime * timeUnits.magnitude);
+    }
```

Comment:
I'm glad there is so much less logic in this component now, but it seems like this last little bit could be moved to a selector. You could also have the selector default the timeUnits.name to "s".

Reply:
Done, check it out!